### PR TITLE
Remove tangle LRU cache

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-tangle/src/vertices.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        RwLockWriteGuard::try_map(table, |table| {
+            table.get_mut(hash, equivalent_id(message_id)).map(|(_, v)| v)
+        })
+        .ok()
+    }
+
+    pub(crate) async fn pop_random(&self) -> Option<Vertex> {
+        let index = rand::thread_rng().gen_range(0..self.tables.len());
+        // SAFETY: `index < self.tables.len()` by construction.
+        let mut table = unsafe { self.tables.get_unchecked(index) }.write().await;
```
Comment (Adam-Gleave) [inline bee-tangle/src/vertices.rs:82]: Would this work here to get rid of the `unsafe`?

```suggestion
        let mut table = self.tables.choose(&mut rand::thread_rng()).write().await;
```

File: bee-tangle/src/tangle.rs [hunk 2]
```diff
@@ -644,62 +607,38 @@ impl<B: StorageBackend> Tangle<B> {
...
+                let mut retries = 1;
+                loop {
+                    if self.vertices.pop_random().await.is_none() {
+                        log::info!("retrying cache eviction (attempt #{})", retries);
+                        retries += 1;
+                    } else {
+                        break;
+                    }
+
+                    if retries > 10 {
```
Comment (Adam-Gleave) [inline bee-tangle/src/tangle.rs]: Maybe make this a constant somewhere.

File: bee-tangle/src/vertices.rs (hunk 1 above)
Comment (thibault-martinez) [reply bee-tangle/src/vertices.rs:82]: I believe the unchecked get was here for performance reason as we know the index is valid. @pvdrz ?

Comment (Adam-Gleave) [reply bee-tangle/src/vertices.rs:82]: Ah that makes sense, I think performance would definitely be better that way.

Comment (pvdrz) [reply bee-tangle/src/vertices.rs:82]: yes, but I wouldn't be surprised if rust can optimize it away, who knows

**Review by pvdrz — DISMISSED** [review]

**Review by Adam-Gleave — DISMISSED** [review]

**Review by pvdrz — APPROVED** [review]

**Review by Adam-Gleave — APPROVED** [review]

**Pull request merged**